src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultTransactionHandler.cs

```diff
@@ -176,4 +161,36 @@ public async Task<GatewayModel.TransactionSubmitResponse> Submit(GatewayModel.Tr
     {
         return await _submissionService.HandleSubmitRequest(request, token);
     }
+
+    public async Task<GatewayModel.StreamTransactionsResponse> StreamTransactions(GatewayModel.StreamTransactionsRequest request, CancellationToken token = default)
+    {
+        var atLedgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
+        var fromLedgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadForwardRequest(request.FromLedgerState, token) ?? atLedgerState;
+
+        var kindFilter = request.KindFilter switch
+        {
+            GatewayModel.StreamTransactionsRequest.KindFilterEnum.All => LedgerTransactionKindFilter.AllAnnotated,
+            GatewayModel.StreamTransactionsRequest.KindFilterEnum.User => LedgerTransactionKindFilter.UserOnly,
+            GatewayModel.StreamTransactionsRequest.KindFilterEnum.EpochChange => LedgerTransactionKindFilter.EpochChangeOnly,
+            null => LedgerTransactionKindFilter.UserOnly,
+            _ => throw new UnreachableException(),
```

Comment:
could we pass argument value that wasn't expected as exception message? That would improve observability and debugging by a lot. I guess it's worth revisiting and fixing all UnreachableException (we have just 7 of them should be quick fix).

Reply:
I've personally mixed feelings about `UnreachableException` I used here and there over good-old `ArgumentOutOfRangeException` but let's discuss that outside of this PR.